pkg/pubsub/operations/subscription.go

```diff
@@ -128,26 +172,49 @@ func (s *SubscriptionOps) Run(ctx context.Context) error {
 		logger.Info("Previously created.")
 
 	case ActionCreate:
+		// Load the topic.
+		topic, err := s.getTopic(ctx)
+		if err != nil {
+			logger.Fatal("Failed to get topic.", zap.Error(err))
+		}
+		// subConfig is the wanted config based on settings.
+		subConfig := pubsub.SubscriptionConfig{
+			Topic:               topic,
+			AckDeadline:         s.AckDeadline,
+			RetainAckedMessages: s.RetainAckedMessages,
+			RetentionDuration:   s.RetentionDuration,
+		}
+
 		// If topic doesn't exist, create it.
 		if !exists {
-			// Load the topic.
-			topic, err := s.getTopic(ctx)
-			if err != nil {
-				logger.Fatal("Failed to get topic.", zap.Error(err))
-			}
 			// Create a new subscription to the previous topic with the given name.
-			sub, err = s.Client.CreateSubscription(ctx, s.Subscription, pubsub.SubscriptionConfig{
-				Topic:             topic,
-				AckDeadline:       30 * time.Second,
-				RetentionDuration: 25 * time.Hour,
-			})
+			sub, err = s.Client.CreateSubscription(ctx, s.Subscription, subConfig)
 			if err != nil {
 				logger.Fatal("Failed to create subscription.", zap.Error(err))
 			}
 			logger.Info("Successfully created.")
 		} else {
 			// TODO: here is where we could update config.
 			logger.Info("Previously created.")
+			// Get current config.
+			currentConfig, err := sub.Config(ctx)
+			if err != nil {
+				logger.Fatal("Failed to get subscription config.", zap.Error(err))
+			}
+			// Compare the current config to the expected config. Update if different.
+			if diff := cmp.Diff(subConfig, currentConfig, ignoreSubConfig); diff != "" {
+				_, err := sub.Update(ctx, pubsub.SubscriptionConfig{
+					AckDeadline:         s.AckDeadline,
+					RetainAckedMessages: s.RetainAckedMessages,
+					RetentionDuration:   s.RetentionDuration,
+					Labels:              currentConfig.Labels,
+				})
+				if err != nil {
+					logger.Fatal("Failed to update subscription config.", zap.Error(err))
```

Comment:
same.

Reply:
This kills the job and marks it failed. working as intended.